tests/test_mbed_targets.py

```diff
@@ -256,58 +202,40 @@ def test_get_target_failure(self, mocked_get_target_data):
         """Check MbedTargets handles queries without a match."""
         mocked_get_target_data.return_value = []
 
-        mbed_targets = MbedTargets.from_offline_database()
+        mbed_targets = MbedTargets.from_online_database()
 
         with self.assertRaises(UnknownTarget):
             mbed_targets.get_target(board_name="unknown product code")
 
-    def test_json_dump(self, mocked_get_target_data):
-        fake_target_data = [
+    @mock.patch("mbed_targets._internal.target_database.get_offline_target_data")
+    def test_json_dump_from_raw_and_fitered_data(self, mocked_get_offline_target_data, mocked_get_online_target_data):
+        raw_target_data = [
             {"attributes": {"product_code": "0200", "board": "test"}},
-            {"attributes": {"product_code": "0100", "board": "test"}},
+            {"attributes": {"product_code": "0100", "board": "test2"}},
         ]
-        mocked_get_target_data.return_value = fake_target_data
-
-        mbed_targets = MbedTargets.from_offline_database()
-        json_str = mbed_targets.json_dump()
-
-        self.assertEqual(
-            json.loads(json_str), fake_target_data, "Deserialised JSON string should match original target data"
-        )
-
-    @mock.patch("mbed_targets._internal.target_database.get_online_target_data")
-    def test_subtracting_targets_takes_difference_non_symmetric(
-        self, mocked_online_target_data, mocked_offline_target_data
-    ):
-        tgt_data_off = _make_dummy_internal_target_data()
-        tgt_data_on = _make_dummy_internal_target_data()
-        tgt_data_off.pop()
-        mocked_offline_target_data.return_value = tgt_data_off
-        mocked_online_target_data.return_value = tgt_data_on
-
-        tgts_off = MbedTargets.from_offline_database()
-        tgts_on = MbedTargets.from_online_database()
-        diff_base_on = tgts_on - tgts_off
-        diff_base_off = tgts_off - tgts_on
+        mocked_get_online_target_data.return_value = raw_target_data
 
-        self.assertEqual(len(diff_base_on), 1)
-        self.assertEqual(len(diff_base_off), 0)
+        targets = [MbedTarget.from_target_entry(t) for t in raw_target_data]
+        filtered_target_data = [target.__dict__ for target in targets]
```

Comment:
Is `__dict__` equivalent to `asdict`? If so, I'd rather we avoid using dunders where possible.

Reply:
`asdict` just calls `object.__dict__` AFAIK. I'll use the `asdict` wrapper in the tests though.